### ISSUE TITLE
Deploy to Openshift throws authentication error while push command does not

### ISSUE THREAD

**Issue by ehelms** [ansible/ansible-container#542]
##### ISSUE TYPE
 - Bug Report

##### container.yml 

```
---
version: "2"
settings:
  conductor_base: centos:7
defaults:
  POSTGRES_USER: foreman
  POSTGRES_PASSWORD: foreman
  POSTGRES_DB: foreman
  POSTGRES_PORT: 5432
  MONGODB_USER: admin
  MONGODB_PASSWORD: admin
  MONGODB_DATABASE: pulp_database
  MONGODB_PORT: 27017
services:
  foreman-base:
    from: centos:7
    roles:
      - epel-repositories
      - role: puppet-repositories
        puppet_repositories_version: 4
      - foreman-repositories
      - katello-repositories
      - foreman
  foreman:
    from: openshift-foreman-base:latest
    command: ['/usr/bin/start-foreman.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASSWORD={{ POSTGRES_PASSWORD }}
      - SEED_ADMIN_USER=admin
      - SEED_ADMIN_PASSWORD=changeme
      - SEED_ORGANIZATION=Default Organization
      - SEED_LOCATION=Raleigh
    ports:
      - 8080:8080
    expose:
      - 8080
  foreman-tasks:
    from: openshift-foreman-base:latest
    roles:
      - foreman-tasks
    command: ['/usr/bin/start-foreman-tasks.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASSWORD={{ POSTGRES_PASSWORD }}
  foreman-proxy:
    from: centos:7
    roles:
      - epel-repositories
      - foreman-repositories
      - foreman-proxy
    command: ['/usr/bin/start-foreman-proxy.sh']
    expose:
      - 8080
  foreman-proxy-register:
    from: centos:7
    roles:
      - epel-repositories
      - foreman-repositories
      - foreman-proxy
    command: ['/usr/bin/register.sh']
    environment:
      - FOREMAN_PROXY_SERVICE_HOST=foreman_proxy
      - FOREMAN_PROXY_SERVICE_PORT=8080
      - FOREMAN_SERVICE_HOST=foreman
      - FOREMAN_SERVICE_PORT=8080
    depends_on:
      - foreman
      - foreman-proxy
  cp-app:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - candlepin
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASSWORD={{ POSTGRES_PASSWORD }}
      - POSTGRES_PORT={{ POSTGRES_PORT }}
      - POSTGRES_SERVICE=postgres
      - QPID_SERVICE=qpid
      - QPID_PORT=5672
    expose:
      - 8080
    command: ['/usr/bin/start_candlepin.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    options:
      kube:
        runAsUser: 1000
  qpid:
    from: centos:7
    roles:
      - epel-repositories
      - qpid
    command: ['/usr/bin/startup.sh']
    expose:
      - 5672
  pulp:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
    command: ['/usr/bin/start_httpd.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    expose:
      - 8080
    depends_on:
      - qpid
      - mongodb
  pulp-worker:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
      - pulp-worker
    entrypoint: ['/usr/bin/entrypoint.sh']
    depends_on:
      - qpid
      - pulp
    depends_on:
      - mongodb
    volumes:
      - pulp-data:/var/lib/pulp
      - puppet-data:/etc/puppet
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: pulp-data
            claim_name: pulp-data
            access_modes:
              - ReadWriteMany
          - volume_name: puppet-data
            claim_name: puppet-data
            access_modes:
              - ReadWriteMany
  pulp-resource-manager:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
      - pulp-resource-manager
    depends_on:
      - qpid
      - pulp
    entrypoint: ['/usr/bin/entrypoint.sh']
  pulp-celerybeat:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
      - pulp-celerybeat
    entrypoint: ['/usr/bin/entrypoint.sh']
    depends_on:
      - qpid
      - pulp
  puppet:
    from: puppet/puppetserver
    volumes:
      - puppet-data:/etc/puppet
    expose:
      - 8140
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: puppet-data
            claim_name: puppet-data
            access_modes:
              - ReadWriteMany
  postgres:
    from: ansible/postgresql:latest
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASS={{ POSTGRES_PASSWORD }}
      - PGDATA=/var/lib/pgsql/data/userdata
    volumes:
      - postgres-data:/var/lib/pgsql/data
    expose:
      - "{{ POSTGRES_PORT }}"
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: postgres-data
            claim_name: postgres-data
            access_modes:
              - ReadWriteMany
  mongodb:
    from: centos/mongodb-26-centos7:latest
    environment:
      - MONGODB_USER={{ MONGODB_USER }}
      - MONGODB_PASSWORD={{ MONGODB_PASSWORD }}
      - MONGODB_DATABASE={{ MONGODB_DATABASE }}
      - MONGODB_ADMIN_PASSWORD={{ MONGODB_PASSWORD }}
    volumes:
      - mongodb-data:/var/lib/mongodb/data
    expose:
      - "{{ MONGODB_PORT }}"
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: mongodb-data
            claim_name: mongodb-data
            access_modes:
              - ReadWriteMany
volumes:
  postgres-data: {}
  mongodb-data: {}
  puppet-data: {}
  pulp-data: {}
registries:
  oc-cluster:
    url: http://172.30.1.1:5000
    namespace: foreman
  # Add optional registries used for deployment. For example:
  #  google:
  #    url: https://gcr.io
  #    namespace: my-cool-project-xxxxxx
```

##### OS / ENVIRONMENT
```
Ansible Container, version 0.9.1rc0
Linux, war.eagle, 4.6.6-200.fc23.x86_64, #1 SMP Wed Aug 10 23:13:35 UTC 2016, x86_64
2.7.11 (default, Jul  8 2016, 19:45:00) 
[GCC 5.3.1 20160406 (Red Hat 5.3.1-6)] /usr/bin/python
{
  "ContainersPaused": 0, 
  "Labels": null, 
  "DefaultRuntime": "runc", 
  "CgroupDriver": "cgroupfs", 
  "ContainersRunning": 6, 
  "NGoroutines": 56, 
  "Swarm": {
    "Managers": 0, 
    "ControlAvailable": false, 
    "NodeID": "", 
    "Cluster": {
      "Spec": {
        "Raft": {}, 
        "CAConfig": {}, 
        "Dispatcher": {}, 
        "Orchestration": {}, 
        "TaskDefaults": {}
      }, 
      "Version": {}, 
      "ID": "", 
      "CreatedAt": "0001-01-01T00:00:00Z", 
      "UpdatedAt": "0001-01-01T00:00:00Z"
    }, 
    "Nodes": 0, 
    "Error": "", 
    "RemoteManagers": null, 
    "LocalNodeState": "inactive", 
    "NodeAddr": ""
  }, 
  "LoggingDriver": "json-file", 
  "OSType": "linux", 
  "HttpProxy": "", 
  "Runtimes": {
    "runc": {
      "path": "docker-runc"
    }
  }, 
  "DriverStatus": [
    [
      "Pool Name", 
      "docker-253:3-17170446-pool"
    ], 
    [
      "Pool Blocksize", 
      "65.54 kB"
    ], 
    [
      "Base Device Size", 
      "10.74 GB"
    ], 
    [
      "Backing Filesystem", 
      "xfs"
    ], 
    [
      "Data file", 
      "/dev/loop0"
    ], 
    [
      "Metadata file", 
      "/dev/loop1"
    ], 
    [
      "Data Space Used", 
      "26.71 GB"
    ], 
    [
      "Data Space Total", 
      "107.4 GB"
    ], 
    [
      "Data Space Available", 
      "80.66 GB"
    ], 
    [
      "Metadata Space Used", 
      "28.62 MB"
    ], 
    [
      "Metadata Space Total", 
      "2.147 GB"
    ], 
    [
      "Metadata Space Available", 
      "2.119 GB"
    ], 
    [
      "Thin Pool Minimum Free Space", 
      "10.74 GB"
    ], 
    [
      "Udev Sync Supported", 
      "true"
    ], 
    [
      "Deferred Removal Enabled", 
      "false"
    ], 
    [
      "Deferred Deletion Enabled", 
      "false"
    ], 
    [
      "Deferred Deleted Device Count", 
      "0"
    ], 
    [
      "Data loop file", 
      "/home/docker/devicemapper/devicemapper/data"
    ], 
    [
      "Metadata loop file", 
      "/home/docker/devicemapper/devicemapper/metadata"
    ], 
    [
      "Library Version", 
      "1.02.109 (2015-09-22)"
    ]
  ], 
  "OperatingSystem": "Fedora 23 (Workstation Edition)", 
  "Containers": 8, 
  "HttpsProxy": "", 
  "BridgeNfIp6tables": true, 
  "MemTotal": 16700760064, 
  "Driver": "devicemapper", 
  "IndexServerAddress": "https://index.docker.io/v1/", 
  "ClusterStore": "", 
  "ExecutionDriver": "", 
  "SystemStatus": null, 
  "OomKillDisable": true, 
  "ClusterAdvertise": "", 
  "SystemTime": "2017-05-24T08:56:00.348733351-04:00", 
  "Name": "war.eagle", 
  "CPUSet": true, 
  "RegistryConfig": {
    "InsecureRegistryCIDRs": [
      "172.30.0.0/16", 
      "127.0.0.0/8"
    ], 
    "IndexConfigs": {
      "docker.io": {
        "Official": true, 
        "Name": "docker.io", 
        "Secure": true, 
        "Mirrors": null
      }
    }, 
    "Mirrors": null
  }, 
  "SecurityOptions": [
    "seccomp"
  ], 
  "ContainersStopped": 2, 
  "NCPU": 8, 
  "NFd": 55, 
  "Architecture": "x86_64", 
  "KernelMemory": true, 
  "CpuCfsQuota": true, 
  "Debug": false, 
  "ID": "PP7J:YWUF:VJG7:OPQQ:AVCZ:BEIA:XEX2:BB7W:ELOD:I2QM:O472:Y2MB", 
  "IPv4Forwarding": true, 
  "KernelVersion": "4.6.6-200.fc23.x86_64", 
  "BridgeNfIptables": true, 
  "NoProxy": "", 
  "LiveRestoreEnabled": false, 
  "ServerVersion": "1.12.6", 
  "CpuCfsPeriod": true, 
  "ExperimentalBuild": false, 
  "MemoryLimit": true, 
  "SwapLimit": true, 
  "Plugins": {
    "Volume": [
      "local"
    ], 
    "Network": [
      "host", 
      "bridge", 
      "null", 
      "overlay"
    ], 
    "Authorization": null
  }, 
  "Images": 163, 
  "DockerRootDir": "/home/docker", 
  "NEventsListener": 0, 
  "CPUShares": true
}
{
  "KernelVersion": "4.6.6-200.fc23.x86_64", 
  "Os": "linux", 
  "BuildTime": "2017-01-10T20:21:11.017972579+00:00", 
  "ApiVersion": "1.24", 
  "Version": "1.12.6", 
  "GitCommit": "78d1802", 
  "Arch": "amd64", 
  "GoVersion": "go1.6.4"
}
```
##### SUMMARY
I had a working build/deployment using the 0.3 version of ansible-container and have been working to update to the latest 0.9. After getting the build steps finalized, I turned to ensuring that my deployment workflow still worked. In the 0.3 era, I would do a `push` and then a `shipit`. So running these same steps, using a local Openshift via `oc cluster` I run into an issue where the push command works for pushing to the Openshift registry but when I run the deploy command I get a failure to authenticate.

##### STEPS TO REPRODUCE
1. Run ansible-container build
2. Using Openshift 1.4+, run oc cluster up to ensure an accessible Openshift
3. Add a 'foreman' project to Openshift
4. Run `ansible-container push --push-to http://172.30.1.1:5000/foreman --username developer --password $(oc whoami -t)`
5. Run `ansible-container --engine openshift deploy --push-to http://172.30.1.1:5000/foreman --username developer --password $(oc whoami -t)`

Note that the push succeeds while the deploy results in:

```
stderr: Parsing conductor CLI args.
Engine integration loaded. Preparing push.	engine=OpenShift?
Traceback (most recent call last):
  File "/usr/bin/conductor", line 11, in <module>
    load_entry_point('ansible-container', 'console_scripts', 'conductor')()
  File "/_ansible/container/__init__.py", line 19, in __wrapped__
    return fn(*args, **kwargs)
  File "/_ansible/container/cli.py", line 355, in conductor_commandline
    **params)
  File "/_ansible/container/__init__.py", line 19, in __wrapped__
    return fn(*args, **kwargs)
  File "/_ansible/container/core.py", line 844, in conductorcmd_push
    username, password = engine.login(username, password, email, url, config_path)
  File "/_ansible/container/__init__.py", line 19, in __wrapped__
    return fn(*args, **kwargs)
  File "/_ansible/container/docker/engine.py", line 780, in login
    u"Error logging into registry: {}".format(exc)
container.exceptions.AnsibleContainerConductorException: Error logging into registry: 401 Client Error: Unauthorized ("Get http://172.30.1.1:5000/v2/: unauthorized: authentication required")
Conductor terminated. Cleaning up.	command_rc=1 conductor_id=b058a79e8510485081085ee4c7d1df05839d561dbd17ec5fe756978d22d11a0b save_container=False
ERROR	Conductor exited with status 1	
```

I use a playbook to do these steps, so for reference here it is:

```
---
- hosts: localhost
  vars:
    registry: oc-cluster
    project: foreman
  tasks:
    - name: 'Get Openshift token'
      command: 'oc whoami -t'
      register: token

    - name: 'Push images to Openshift'
      command: "ansible-container push --push-to {{ registry }}/{{ project }} --username developer --password {{ token.stdout }}"

    - name: 'Generate deployment role'
      command: "ansible-container --engine openshift deploy --push-to {{ registry }} --username developer --password {{ token.stdout }}"

    - name: 'Deploy to Openshift'
      command: "ansible-playbook ansible/shipit-openshift.yml"
      register: shipit_output
      ignore_errors: true

    - debug:
        msg: "{{ shipit_output.stdout.split('\n') }}"
```

**Comment by ehelms** [comment]
@chouseknecht I can confirm, using your fork/PR that the issue was fixed for me! Thanks! Feel free to close this once that PR is merged.

**Comment by j00bar** [comment]
@ehelms Awesome dogfooding, mate! Thanks so much!
